### PR TITLE
1.72 toolchain, 1.66 MSRV

### PR DESCRIPTION
**Pull request by robertbastian**
This should be reviewed commit-by-commit.

### CODE REVIEW

File: utils/zerovec/src/map/borrowed.rs [hunk 1]
```diff
@@ -63,10 +63,7 @@ where
     V: ?Sized,
 {
     fn clone(&self) -> Self {
-        ZeroMapBorrowed {
-            keys: self.keys,
-            values: self.values,
-        }
+        *self
```
Comment (robertbastian) [inline utils/zerovec/src/map/borrowed.rs:66]: @Manishearth

File: provider/datagen/src/baked_exporter.rs [hunk 2]
```diff
@@ -328,14 +330,14 @@ impl DataExporter for BakedExporter {
         let bake = payload.tokenize(&self.dependencies);
 
         self.write_impl_macro(quote! {
-            #[clippy::msrv = "1.65"]
+            #[clippy::msrv = #MSRV]
```
Comment (Manishearth) [inline provider/datagen/src/baked_exporter.rs:333]: praise: clever

**Review by Manishearth — APPROVED** [review]
r+ for actual changes. I wasn't sure if we had decided to do this bump, AIUI our MSRV policy was to only bump when we have a reason to, yes?

**Comment by robertbastian** [comment]
This has convenience functions on BTreeSet/map that might be ... convenient.

I'm also happy to wait, but I definitely want to do 1.67 when we're allowed to, because that has `ilog10`.

**Review by sffc — APPROVED** [review]

**Comment by sffc** [comment]
I'd like to use `core::hint::black_box` in a few places, and I think there are some places where the `add_unsigned` functions also would be useful. Also I _think_ https://github.com/rust-lang/rust/pull/101520 shipped in 1.66 which will help us remove some unsafe in the yoke crate.

**Comment by Manishearth** [comment]
that works for me!

**Pull request merged**